config/routes.rb

```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root "users#index"
+
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+
+  get "/users/:user_id/posts", to: "user_posts#index"
+  get "/users/:user_id/posts/:id", to: "user_posts#show"
```

Comment:
- [ ] I suggest that you use `Resource routing` because it allows you to quickly declare all of the common routes for a given resourceful controller. you can simply do:
```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```